torchtune/training/_distributed.py

```diff
@@ -546,3 +564,72 @@ def shard_model(
 
     # Finally shard the entire model to account for any stragglers
     fully_shard(model, **fsdp_kwargs)
+
+
+def get_tp_plan(model_type: str) -> Dict[str, ParallelStyle]:
+    """
+    Get the TP plan for a given model type.
+
+    Args:
+        model_type (str): The model type to get the TP plan for.
+
+    Returns:
+        Dict[str, str]: A dictionary mapping layer names to their corresponding TP plan.
+
+    Raises:
+        ValueError: If trying to get the TP plan for non llama models.
+    """
+    # For now, we only support base TP plan, will add more plan later
+    if model_type == "LLAMA3_VISION":
+        from torchtune.models.llama3_2_vision._model_builders import LLAMA_DEEP_FUSION_VISION_TP_PLAN
+
+        return LLAMA_DEEP_FUSION_VISION_TP_PLAN
+    elif model_type == "LLAMA3":
+        return BASE_LLAMA_TP_PLAN
+    else:
+        raise ValueError("TP is only supported for llama type models right now.")
+
+
+def adjust_attention_for_tp(
+    model: nn.Module,
+    tp_mesh: DeviceMesh,
+) -> nn.Module:
+    """
+    Adjusts the number of attention heads and dimension in the model to account for tensor parallelism.
```

Comment:
This description is too vague. We should communicate what exactly is happening to the users.